HCP docs update for IBM Power (#92359)
openshift/openshift-docs: modules/hcp-create-infraenv.adoc

// Module included in the following assemblies:
//
// * hosted_control_planes/hcp-deploy/hcp-deploy-ibm-power.adoc

:_mod-docs-content-type: PROCEDURE
[id="hcp-create-infraenv_{context}"]
= Create Infraenv

> For heterogeneous node pools, you must create the `infraenv` custom resource (CR) for each architecture. For example, for node pools with `x86_64` and `ppc64le` architectures, create the `infraenv` CR for `x86_64` and `ppc64le`.

Reviewer: If this is required for heterogeneous node pools, should this be a step in the _Creating heterogeneous node pools_ procedure so that people don't miss it?

.Procedure

. Create the `infraenv` resource for heterogeneous node pools by running the following command:
+
[source,terminal]
----
$ export SSH_PUB_KEY=$(cat $HOME/.ssh/id_rsa.pub)
----
+
[source,yaml]
----
envsubst <<"EOF" | oc apply -f -
apiVersion: agent-install.openshift.io/v1beta1
kind: InfraEnv
metadata:
  name: ${HOSTED_CLUSTER_NAME}-${ARCH_X86}
  namespace: ${HOSTED_CONTROL_PLANE_NAMESPACE}
spec:
  cpuArchitecture: ${ARCH_X86}
  pullSecretRef:
    name: pull-secret
  sshAuthorizedKey: ${SSH_PUB_KEY}
EOF
----
+
[source,yaml]
----
envsubst <<"EOF" | oc apply -f -
apiVersion: agent-install.openshift.io/v1beta1
kind: InfraEnv
metadata:
  name: ${HOSTED_CLUSTER_NAME}-${ARCH_PPC64LE}
  namespace: ${HOSTED_CONTROL_PLANE_NAMESPACE}
spec:
  cpuArchitecture: ${ARCH_PPC64LE}
  pullSecretRef:
    name: pull-secret
  sshAuthorizedKey: ${SSH_PUB_KEY}
EOF
----
+

> This creates two `infraenv` resources with `x86_64` and `ppc64le` architectures. Before proceeding, make sure that the OS images for both architectures are added to the `AgentServiceConfig` resource. After this you can use the above infraenvs to get the minimal iso and follow the `create-agent-cluster.md` to add them to your cluster as agents.

Reviewer: > make sure that the OS images for both architectures are added

This procedure adds the images?

Reviewer: ```suggestion
These commands creates two `infraenv` resources with `x86_64` and `ppc64le` architectures. Before proceeding, make sure that the OS images for both architectures are added to the `AgentServiceConfig` resource. After this, you can use the above `infraenv` resources to get the minimal ISO image. Then, follow the `create-agent-cluster.md` to add them to your cluster as agents.
```

Reviewer: >This creates two `infraenv` resources with `x86_64` and `ppc64le` architectures. Before proceeding, make sure that the OS images for both architectures are added to the `AgentServiceConfig` resource. After this you can use the above infraenvs to get the minimal iso and follow the `create-agent-cluster.md` to add them to your cluster as agents.

* Do users know how to verify that the OS was added? Give steps or an example? 
* Do users know how to use an `infraenvs` object?
* _get the minimal ISO image_ Does the `infraenvs` object always get the minimal ISO, or I should make sure to get the minimal? 
* What is the `create-agent-cluster.md`?

Reviewer 2: > > This creates two `infraenv` resources with `x86_64` and `ppc64le` architectures. Before proceeding, make sure that the OS images for both architectures are added to the `AgentServiceConfig` resource. After this you can use the above infraenvs to get the minimal iso and follow the `create-agent-cluster.md` to add them to your cluster as agents.
> 
> * Do users know how to verify that the OS was added? Give steps or an example?
> * Do users know how to use an `infraenvs` object?
> * _get the minimal ISO image_ Does the `infraenvs` object always get the minimal ISO, or I should make sure to get the minimal?
> * What is the `create-agent-cluster.md`?

1. Example is added for user to verify the created custom resource `Infraenv`.
2. Yes, users would be knowing it. Also statements are added in the patch which mentions usage of `infraenvs` object to get live iso url.
3. `Infraenv` generates minimal iso. Nothing is to be done from user side to make it minimal.
4. Removed references to `create-agent-cluster.md` and added further steps in the recent patch.

Reviewer 2: > > make sure that the OS images for both architectures are added
> 
> This procedure adds the images?

This procedure adds references of OCP images in `AgentServiceConfig` custom resource which is then used by `Infraenv` to generate minimal iso.

// Module included in the following assemblies:
//
// * hosted_control_planes/hcp-deploy/hcp-deploy-ibm-power.adoc

:_mod-docs-content-type: PROCEDURE
[id="hcp-create-infraenv_{context}"]

> = Create Infraenv

Reviewer 3: Create needs to be Creating
What is "Infraenv"? This needs one of the following changes:

```suggestion
= Creating infrastructure environment resources
```
(just a random guess what infraenv might be short for)

```suggestion
= Creating `infraenv` custom resources
```

> For heterogeneous node pools, you must create the `infraenv` custom resource (CR) for each architecture. For example, for node pools with `x86_64` and `ppc64le` architectures, create the `infraenv` CR for `x86_64` and `ppc64le`.

Reviewer 3: ```suggestion
For heterogeneous node pools, you must create an `infraenv` custom resource (CR) for each architecture. For example, for node pools with `x86_64` and `ppc64le` architectures, create an `infraenv` CR for `x86_64` and `ppc64le`.
```

.Procedure

> . Create the `infraenv` resource for heterogeneous node pools by running the following command:

Reviewer 3: This is two commands. Each one should have a step, and the step needs to explain what each does.

+

> [source,yaml]

Reviewer 3: ```suggestion
[source,terminal]
```

----

> envsubst <<"EOF" | oc apply -f -

Reviewer 3: ```suggestion
$ envsubst <<"EOF" | oc apply -f -
```

// Module included in the following assemblies:
//
// * hosted_control_planes/hcp-deploy/hcp-deploy-ibm-power.adoc

:_mod-docs-content-type: PROCEDURE
[id="hcp-create-infraenv_{context}"]
= Creating infrastructure environment resources

For heterogeneous node pools, you must create an `infraenv` custom resource (CR) for each architecture. For example, for node pools with `x86_64` and `ppc64le` architectures, create an `infraenv` CR for `x86_64` and `ppc64le`.

.Procedure

> Before proceeding, make sure that the operating system (OS) images for both `x86_64` and `ppc64le` architectures are added to the `AgentServiceConfig` resource. After this, you can use the `infraenv` resources to get the minimal ISO image.

Reviewer 4: We generally don't put content between ".Procedure" and the first step. So line 13 could be moved above `.Procedure." If it's really important you could make it a note or an IMPORTANT admonition. 

https://redhat-documentation.github.io/modular-docs/#con-creating-procedure-modules_writing-mod-docs

<3> The hosted control plane namespace.
<4> The ssh public key.

. Create the `infraenv` resource with `ppc64le` architecture for heterogeneous node pools by running the following command:
+
[source,yaml]
----
envsubst <<"EOF" | oc apply -f -
apiVersion: agent-install.openshift.io/v1beta1
kind: InfraEnv
metadata:
  name: <hosted_cluster_name>-<arch_ppc64le> <1> <2>
  namespace: <hosted_control_plane_namespace> <3>
spec:
  cpuArchitecture: <arch_ppc64le>
  pullSecretRef:
    name: pull-secret
  sshAuthorizedKey: <ssh_pub_key> <4>
EOF
----
+
<1> The hosted cluster name.
<2> The `ppc64le` architecture.
<3> The hosted control plane namespace.
<4> The ssh public key.

[id="hcp-create-infraenv-verifying_{context}"]
== Verifying the resource creation

. To verify that the `infraenv` resources are created successfully run the following commands:

** Verify that the `x86_64` `infraenv` resource is created successfully:
+
[source,terminal]
----
$ oc describe InfraEnv <hosted_cluster_name>-<arch_x86>
----
+
** Verify that the `ppc64le` `infraenv` resource is created successfully:
+
[source,terminal]
----
$ oc describe InfraEnv <hosted_cluster_name>-<arch_ppc64le>
----

> . Generate a live ISO that allows machines either a virtual machine or a bare-metal machine to join as agents by running the following commands:

Reviewer 4: I *THINK* we can omit "machines" here: 

```suggestion
. Generate a live ISO that allows either a virtual machine or a bare-metal machine to join as agents by running the following commands:
```

+

> . Generate a live ISO for `x86_64`:

Reviewer 4: Maybe 

```suggestion
.. Generate a live ISO for `x86_64`:
```

+
[source,terminal]
----
$ oc -n <hosted_control_plane_namespace> get InfraEnv <hosted_cluster_name>-<arch_x86> -ojsonpath="{.status.isoDownloadURL}"
----
+
. Generate a live ISO for `ppc64le`:
+
[source,terminal]
----
$ oc -n <hosted_control_plane_namespace> get InfraEnv <hosted_cluster_name>-<arch_ppc64le> -ojsonpath="{.status.isoDownloadURL}"

> ----

Reviewer: What do these steps do? Are they actually verification tasks? 
If not, should this be a separate module?
